### ISSUE TITLE
Maybe constructors does not provides concat

### ISSUE THREAD

**Issue by netoguimaraes** [origamitower/folktale#124]
It seems that Maybe constructors does not provides `concat` on 2.0.0-beta1

```javascript 
> var { Just, Nothing }  = require('folktale/data/maybe')                                                                                          
undefined 
> var concat = require('folktale/core/fantasy-land/concat')                                                                                         
undefined                                                                                                                                        
> concat(Just('example'), Just('example'))                                                                                                           
TypeError: folktale:Data.Maybe.Just({ value: "example" }) does not have a method 'concat'.                                                           
    at /home/neto/engProjects/spd/src/node_modules/folktale/helpers/unsupported-method.js:14:11                                                      
    at concat (/home/neto/engProjects/spd/src/node_modules/folktale/core/fantasy-land/concat.js:38:174)                                              
```
### Environment

  - OS: Linux
  - JavaScript VM: V8
  - Folktale version: 2.0.0-beta1

**Comment by robotlolita** [comment]
Yeah, right now those are still not implemented :(

Semigroup and some other algebras are [planned for a post-2.0 release](http://origamitower.github.io/folktale/en/guides.do-folktale-structures-implement-fantasy-land-.html#folktale-2), though. And since the only thing blocking an actual 2.0 release right now is me finishing the new website, this should happen soon(ish).

**Comment by netoguimaraes** [comment]
Okay, thank you by the feedback. Actually we are using a monkey patch in our project to temporarily solve this. But we can may help you with this. I will talk with my team about it.

**Comment by diasbruno** [comment]
Hi @robotlolita, while implementing `semigroup` and `monoid` for `maybe`, this problem happen when running the `laws` for `monoid`.

```js
Exception: type[empty] is not a function.
```

[Here is the patch. github/diasbruno/folktale](https://github.com/origamitower/folktale/compare/master...diasbruno:feature/maybe-concat)

Probably, something is missing to inject `.empty` on the type `Maybe`.

**Comment by diasbruno** [comment]
Oh yeah, I got why `mempty` can't be implemented. It depends on the `a` on `Monoid a => a` to implement it correctly.

**Comment by robotlolita** [comment]
@netoguimaraes I'd happily accept a PR for adding Semigroup to Maybe :)


@diasbruno hey~☆

That was a bug in the test. https://github.com/origamitower/folktale/commit/268af9e775a8038c2e48733a6418d4bb80adf5f1 should fix it.

**Comment by diasbruno** [comment]
Thanks, I'll check it out.

**Comment by diasbruno** [comment]
That fix the problem. Thank you.

[github/diasbruno/folktale#feature/maybe-semigroup-monoid](https://github.com/origamitower/folktale/compare/master...diasbruno:feature/maybe-semigroup-monoid)

**Comment by netoguimaraes** [comment]
@robotlolita 
@diasbruno and I are working on the same project, that need this implementation. So this PR will allow us to remove the actual monkey patch ;). Thank you for the atention, by the way <3;

**Comment by robotlolita** [comment]
I'll try to release this during the weekend :)

**Comment by robotlolita** [comment]
Oh, I forgot to comment, but this has been released as [v2.0.0-rc1](https://github.com/origamitower/folktale/releases/tag/release%2Fv2.0.0-rc1), though that release contains some breaking changes.
